HiP-UserStore/Migrations/Migration1PropertyChanged.cs

```diff
@@ -0,0 +1,131 @@
+﻿using PaderbornUniversity.SILab.Hip.EventSourcing;
+using PaderbornUniversity.SILab.Hip.EventSourcing.Events;
+using PaderbornUniversity.SILab.Hip.EventSourcing.Migrations;
+using PaderbornUniversity.SILab.Hip.UserStore.Model.Entity;
+using PaderbornUniversity.SILab.Hip.UserStore.Model.Events;
+using PaderbornUniversity.SILab.Hip.UserStore.Model.Rest;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace PaderbornUniversity.SILab.Hip.UserStore.Migrations
+{
+    [StreamMigration(from: 0, to: 1)]
+    public class Migration1PropertyChanged : IStreamMigration
+    {
+        private Dictionary<(ResourceType, int), object> argumentDictionary = new Dictionary<(ResourceType, int), object>();
+
+        public async Task MigrateAsync(IStreamMigrationArgs e)
+        {
+            var events = e.GetExistingEvents();
+            DateTimeOffset timestamp;
+
+            while (await events.MoveNextAsync())
+            {
+                var currentEvent = events.Current;
+                IEnumerable<PropertyChangedEvent> propEvents = new List<PropertyChangedEvent>();
+
+                if (currentEvent is UserActivityBaseEvent)
+                {
+                    switch (currentEvent)
+                    {
+                        case UserCreated ev:
+                            var emptyUserArgs = new UserArgs2();
+                            e.AppendEvent(new CreatedEvent(ev.GetEntityType().Name, ev.Id, ev.UserId)
+                            {
+                                Timestamp = ev.Timestamp
+                            });
+                            var newUserArgs = new UserArgs2 
+                            {
+                                FirstName = ev.Properties.FirstName,
```

Comment:
If have tested your migration with the current data from the UserStore. Apparently there are events where the Properties are set to null(?). Also as I have seen it might be possible that FirstName and/or LastName are set to null. The migration should be able to handle that